jinahub/rankers/DPRReaderRanker/tests/unit/test_ranker.py

```diff
@@ -0,0 +1,176 @@
+from itertools import chain
+from pathlib import Path
+from typing import List
+
+import pytest
+import torch
+from jina import Document, DocumentArray
+from jina.executors import BaseExecutor
+
+from ...dpr_reader import DPRReaderRanker
+
+
+@pytest.fixture(scope='session')
+def basic_ranker() -> DPRReaderRanker:
+    return DPRReaderRanker()
+
+
+@pytest.fixture(scope='session')
+def basic_ranker_title() -> DPRReaderRanker:
+    return DPRReaderRanker(title_tag_key='title')
+
+
+@pytest.fixture(scope='function')
+def example_docs(request) -> DocumentArray:
+    """Generate example documents with matches (up to 10)."""
+    data_dir = Path(__file__).parent.parent / 'test_data'
+    with open(data_dir / 'questions.txt', 'r') as f:
+        questions = [line.strip() for line in f.readlines()]
+
+    with open(data_dir / 'text.txt', 'r') as f:
+        passages = [line.strip() for line in f.readlines()]
+
+    with open(data_dir / 'titles.txt', 'r') as f:
+        titles = [line.strip() for line in f.readlines()]
+
+    docs = DocumentArray()
+    for question in questions[: request.param]:
+        doc = Document(text=question)
+        matches = [
+            Document(text=passage, tags={'title': title})
+            for passage, title in zip(passages[:10], titles)
+        ]
+        doc.matches.extend(matches)
+        docs.append(doc)
```

Comment:
Add some randomness to show the reranking actually works.
```suggestion
        doc.matches.shuffle(seed=0)
        docs.append(doc)
```

Reply:
There is already a test that checks that proper matches get to the first place (and matches are constructed from scratch anyways), so it's not needed